### PR TITLE
Add a countermeasure against sql injections

### PR DESCRIPTION
**Pull request by mrsupiri**
lookupByName function had a security flow that allowed people to inject sql queries to name and execute remote queries via b3 to mysql database. I added a simple escape_string function and remove injections before it send to the database to process.

### CODE REVIEW

File: b3/clients.py [hunk 1]
```diff
@@ -1532,6 +1533,20 @@ def getByCID(self, cid):
                 return None
         return None
 
+    def escape_string(value, mapping=None):
+        if isinstance(value, unicode):
+            return _escape_unicode(value)
+        assert isinstance(value, (bytes, bytearray))
```
Comment (danielepantaleone) [inline b3/clients.py]: I would not put an assertion in code that is going straight in production. Maybe just return None if the value is not acceptable and handle the case where the escaped string is None.

File: b3/clients.py [hunk 2]
```diff
@@ -1532,6 +1533,20 @@ def getByCID(self, cid):
                 return None
         return None
 
+    def escape_string(value, mapping=None):
+        if isinstance(value, unicode):
+            return _escape_unicode(value)
```
Comment (danielepantaleone) [inline b3/clients.py]: Where is this `_escape_unicode ` function defined?

Comment (mrsupiri) [reply b3/clients.py]: oh I didn't push to tested code :/ I will fix it asap I am kind of new to these version control things sorry

**Comment by danielepantaleone** [comment]
Thank you. I would have to trust you on this since I have no way to test it at the moment.

**Comment by mrsupiri** [comment]
I tested it just now on cod4, unicode string escape also work well. I am still waiting to travis to complete building hope fully it will pass too 👍

**Comment by danielepantaleone** [comment]
The Travis CI build will not pass because it's broken since months. However, I'm not going to fix it since I'm rewriting B3 core from scratch and updating automated tests for an old codebase it's pointless (although I don't know how much time I need to complete the new codebase) 😁

**Comment by mrsupiri** [comment]
I am using b3 for like 4-5 years now. So I like to help you. let me know if there anything I can do. 😄 😃

**Comment by danielepantaleone** [comment]
Once I push the "new" codebase it will be plenty of things to do, don't worry 😄

**Comment by mrsupiri** [comment]
he he i am waiting 😉🤗

**Pull request merged**